### PR TITLE
Code generation to support bulk APIs

### PR DESCRIPTION
**Pull request by DanG100**
* Support bulk_create
* Support enum values that are assign to custom values
* Support setting the hostif name separately

### CODE REVIEW

File: dataplane/standalone/apigen/ccgen/ccgen.go [hunk 1]
```diff
@@ -42,11 +42,17 @@ func Generate(doc *docparser.SAIInfo, sai *saiast.SAIAPI) (map[string]string, er
 		if apiName == "switch" {
 			ccData.Globals = append(ccData.Globals, "std::unique_ptr<PortStateReactor> port_state;")
 		}
+		if apiName == "hostif" {
+			ccData.Globals = append(ccData.Globals, "int nextIdx = 1;")
```
Comment (wenovus) [inline dataplane/standalone/apigen/ccgen/ccgen.go:46]: nit: use switch

File: dataplane/standalone/apigen/ccgen/ccgen.go [hunk 2]
```diff
@@ -71,19 +77,26 @@ func sanitizeProtoName(inName string) string {
 }
 
 // createCCData returns a struct with the template data for the given function.
-func createCCData(meta *saiast.FuncMetadata, apiName string, sai *saiast.SAIAPI, info *docparser.SAIInfo, fn *saiast.TypeDecl) *templateFunc {
+func createCCData(meta *saiast.FuncMetadata, apiName string, sai *saiast.SAIAPI, info *docparser.SAIInfo, fn *saiast.TypeDecl) (*templateFunc, *templateFunc) {
```
Comment (wenovus) [inline dataplane/standalone/apigen/ccgen/ccgen.go:80]: Update the comment to say what the two functions are each for?

File: dataplane/standalone/apigen/docparser/docparser.go [hunk 3]
```diff
@@ -29,7 +30,12 @@ type SAIInfo struct {
 	// attrs is a map from sai type (sai_port_t) to its attributes.
 	Attrs map[string]*Attr
 	// attrs is a map from enum name (sai_port_media_type_t) to the values of the enum.
```
Comment (wenovus) [inline dataplane/standalone/apigen/docparser/docparser.go:32]: attrs -> Enums

File: dataplane/standalone/apigen/docparser/docparser.go [hunk 4]
```diff
@@ -165,10 +171,30 @@ func memberToAttrInfo(enum MemberDef) *Attr {
...
 
-func memberToEnumValueStrings(enum MemberDef) []string {
-	res := []string{}
-	for _, value := range enum.EnumValues {
-		res = append(res, value.Name)
+func memberToEnumValueStrings(enum MemberDef) []*Enum {
+	res := []*Enum{}
+	prev := -1
+	for i, value := range enum.EnumValues {
+		val := prev + 1
```
Comment (wenovus) [inline dataplane/standalone/apigen/docparser/docparser.go:178]: nit: perhaps `next := 0` is slightly better. Perhaps it's just I'm not sure what "prev/next" is just by scanning this.

File: dataplane/standalone/apigen/saiast/saiast.go [hunk 5]
```diff
@@ -126,12 +126,14 @@ func (sai *SAIAPI) GetFuncMeta(fn *TypeDecl) *FuncMetadata {
 		if strings.Contains(param.Typ, "entry") {
 			meta.Entry = TrimSAIName(strings.TrimPrefix(param.Typ, "const "), true, false)
 		}
-		if i == 1 && param.Name == "switch_id" {
+		if (i == 0 || i == 1) && param.Name == "switch_id" {
```
Comment (wenovus) [inline dataplane/standalone/apigen/saiast/saiast.go:129]: Perhaps comment on the significance of `(i == 0 || i == 1)`

**Review by wenovus — APPROVED** [review]

File: dataplane/standalone/apigen/docparser/docparser.go (hunk 4 above)
Comment (DanG100) [reply dataplane/standalone/apigen/docparser/docparser.go]: kept the same, but added comment

**Pull request merged**